src/conn/conn_log.c

```diff
@@ -139,30 +134,21 @@ __logmgr_version(WT_SESSION_IMPL *session, bool reconfig)
         return (0);
 
     /*
-     * Set the log file format versions based on compatibility versions set in the connection. We
-     * must set this before we call log_open to open or create a log file.
+     * Set the log file format versions based on compatibility versions set in the connection. The
+     * compatibility version must be set at this point. We must set this before we call log_open to
+     * open or create a log file.
      *
      * Note: downgrade in this context means the new version is not the latest possible version. It
      * does not mean the direction of change from the release we may be running currently.
      */
-    if (conn->compat_major == WT_LOG_V5_MAJOR) {
-        new_version = WT_LOG_VERSION;
-        first_record = WT_LOG_END_HEADER + log->allocsize;
-        downgrade = false;
-    } else if (conn->compat_major == WT_LOG_V4_MAJOR) {
-        if (conn->compat_minor == WT_LOG_V4_MINOR)
-            new_version = 4;
-        else if (conn->compat_minor > WT_LOG_V2_MINOR)
-            new_version = 3;
-        else
-            new_version = 2;
+    WT_ASSERT(session, __wt_version_defined(conn->compat_version));
+    __logmgr_set_log_version(conn->compat_version, &new_version);
+
+    downgrade = new_version != WT_LOG_VERSION;
```

Comment:
Might not be necessary to set it here, `downgrade` is used further down. Would be better to have this line where it is used.

Reply:
Fixed in cea28f3